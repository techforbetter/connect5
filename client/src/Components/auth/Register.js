import React, { Component } from "react";
import axios from "axios";
import {
  Heading, Container, Form, Input, Error,
} from "./PageElementsAuth";

class Register extends Component {
  constructor() {
    super();
    this.state = {
      firstName: "",
      lastName: "",
      email: "",
      password: "",
      password2: "",
      errors: {},
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  // sumbmit to send object to backend to be stored in DB
  onSubmit(e) {
    e.preventDefault();
    const newTrainer = {
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      email: this.state.email,
      password: this.state.password,
      password2: this.state.password2,
    };
    this.registerUser(newTrainer);
  }

  registerUser = (trainerData) => {
    axios
      .post("/register", trainerData)
      .then(res => this.props.history.push("/trainer/login")) // redirect after reguster success
      .catch(err => this.setState({
        errors: err.response.data,
      }));
  };

  render() {
    const { errors } = this.state;
    return (
      <Container>
        <Heading>Create Your Account</Heading>
        <Form noValidate onSubmit={this.onSubmit}>
          <Input
            type="text"
            placeholder="First Name"
            name="firstName"
            value={this.state.firstName}
            onChange={this.onChange}
          />
          {errors.firstName && <Error className="invalid-feedback">{errors.firstName}</Error>}
          <Input
            type="text"
            placeholder="Last Name"
            name="lastName"
            value={this.state.lastName}
            onChange={this.onChange}
          />
          {errors.lastName && <Error className="invalid-feedback">{errors.lastName}</Error>}
          {" "}
          <Input
            type="email"
            placeholder="Email Address"
            name="email"
            value={this.state.email}
            onChange={this.onChange}
          />
          {errors.email && <Error className="invalid-feedback">{errors.email}</Error>}
          <Input
            type="password"
            placeholder="Password"
            name="password"
            value={this.state.password}
            onChange={this.onChange}
          />
          {errors.password && <Error className="invalid-feedback">{errors.password}</Error>}
          <Input
            type="password"
            placeholder="Confirm Password"
            name="password2"
            value={this.state.password2}
            onChange={this.onChange}
          />
          {errors.password2 && <Error className="invalid-feedback">{errors.password2}</Error>}
          <Input type="submit" submit/>
        </Form>
      </Container>
    );
  }
}

export default Register;
