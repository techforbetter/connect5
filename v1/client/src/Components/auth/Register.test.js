import React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router-dom";

import Register from "./Register";

test("Register matches snapshot", () => {
  const context = {};
  const component = renderer.create(
    <StaticRouter context={context}>
      <Register />
    </StaticRouter>,
  );
  expect(component).toMatchSnapshot();
});
