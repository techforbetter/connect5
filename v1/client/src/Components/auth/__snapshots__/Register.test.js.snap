// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register matches snapshot 1`] = `
<div
  className="sc-bdVaJa jvBJEN"
>
  <h1
    className="sc-bwzfXH hIJRJv"
  >
    Create Your Account
  </h1>
  <form
    className="sc-htpNat caZIjQ"
    noValidate={true}
    onSubmit={[Function]}
  >
    <input
      className="sc-bxivhb jCrkAs"
      name="firstName"
      onChange={[Function]}
      placeholder="First Name"
      type="text"
      value=""
    />
    <input
      className="sc-bxivhb jCrkAs"
      name="lastName"
      onChange={[Function]}
      placeholder="Last Name"
      type="text"
      value=""
    />
     
    <input
      className="sc-bxivhb jCrkAs"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      type="email"
      value=""
    />
    <input
      className="sc-bxivhb jCrkAs"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="sc-bxivhb jCrkAs"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value=""
    />
    <input
      className="sc-bxivhb jxQDzy"
      type="submit"
    />
  </form>
</div>
`;
